provider/core.rb

```diff
@@ -397,28 +397,8 @@ def build_object_data(object, output_folder, version)
       )
     end
 
-    def build_url(url_parts, extra = false)
-      (product_url, obj_url) = url_parts
-      extra_arg = ''
-      extra_arg = ', extra_data' if extra
-      ['URI.join(',
-       indent([quote_string(product_url) + ',',
-               'expand_variables(',
-               indent(format_expand_variables(obj_url), 2),
-               indent('data' + extra_arg, 2),
-               ')'], 2),
-       ')'].join("\n")
-    end
-
-    # TODO(rileykarson): Rehome this function.
-    # For some reason the corresponding quote_string function lives in compile/core.rb
-    # and no beside this function.
-    def unquote_string(value)
-      return value.gsub(/"(.*)"/, '\1') if value.start_with?('"')
-      return value.gsub(/'(.*)'/, '\1') if value.start_with?("'")
-
-      value
-    end
+    # This is used within Terraform and Ansible but they implement their own
+    def build_url(url_parts, extra = false) end
```

Comment:
Should we throw an error instead of failing silently if a provider hasn't implemented this and it gets called?

Reply:
Should we prefer raising an error, or I could just remove this base definition?

I'm leaning towards just removing it from here and giving each provider the freedom to do whatever they want

Reply:
I want to split some common mixins out of core.rb in the medium term including this, so I'm slightly for errors for now.